frontend_tests/node_tests/pm_list_data.js

```diff
@@ -0,0 +1,240 @@
+    const num_unread_for_person = 1;
+
+    // initially we append 2 conversations and check for the `convos_to_be_shown` returned in list_info .
+    pm_conversations.recent.insert([101, 102], timestamp);
+    pm_conversations.recent.insert([103], timestamp);
+
+    override(unread, "num_unread_for_person", () => num_unread_for_person);
+
+    list_info = get_list_info();
+
+    const expected_list_info = [
+        {
+            is_active: false,
+            is_group: false,
+            is_zero: false,
+            recipients: "Aaron",
+            status_emoji_info: {
+                emoji_code: 20,
+            },
+            unread: 1,
+            url: "#narrow/pm-with/103-aaron",
+            user_circle_class: "user_circle_empty",
+            user_ids_string: "103",
+        },
+        {
+            recipients: "Alice, Bob",
+            user_ids_string: "101,102",
+            unread: 1,
+            is_active: false,
+            url: "#narrow/pm-with/101,102-group",
+            user_circle_class: undefined,
+            status_emoji_info: undefined,
+            is_group: true,
+            is_zero: false,
+        },
+    ];
+
+    assert.deepEqual(list_info, {
+        convos_to_be_shown: expected_list_info,
+        more_convos_unread_count: 0,
+    });
+
+    // Now we need to test that until the number of conversations reach `max_convos_to_show_with_unreads`
+    // we have all those conversations with unreads shown in default view itself.
+
+    // To check this we go with checking the number of unreads present in `more conversations` li-item.
+    pm_conversations.recent.insert([104], timestamp);
+    pm_conversations.recent.insert([103, 104], timestamp);
+    pm_conversations.recent.insert([105], timestamp);
+    pm_conversations.recent.insert([104, 105], timestamp);
+    pm_conversations.recent.insert([106], timestamp);
+    pm_conversations.recent.insert([105, 106], timestamp);
+
+    // We added a total of 8 conversations, which is the value of `max_convos_to_show_with_unreads`
+    // so even now, the number of unreads in `more conversations` li-item should be 0.
+    list_info = get_list_info();
+
+    assert.deepEqual(list_info.convos_to_be_shown.length, 8);
+    assert.deepEqual(list_info.more_convos_unread_count, 0);
+
+    // Now after adding one more PM, the length in default view would be 9 which would
+    // exceed `max_convos_to_show_with_unreads`, so the number of unreads present in
+    // `more conversations` li-item should be 1 and also the length of `convos_to_be_shown` should be 8.
+    pm_conversations.recent.insert([107], timestamp);
+    list_info = get_list_info();
+    assert.deepEqual(list_info.convos_to_be_shown.length, 8);
+    assert.deepEqual(list_info.more_convos_unread_count, 1);
+
+    // Now, here we test the functionality when the list is zoomed, now according to the logic
+    // we should have all the 9 conversations visible to us.
+    const zoomed = true;
```

Comment:
This variable seems unnecessary.

Reply:
Ahh, i think it is necessary; 
as what i'm testing here is -> I'm sending `zoomed` as `true` in the function `get_list_info` so that now i've all the conversations appended in the list `convos_to_be_shown`;

I've also tried to add a comment for this on the top of this variable;